### PR TITLE
AirPlay/AirTunes: don't dynamically open libraries

### PR DESCRIPTION
**Pull request by lrusak**
We can just link the libraries in, we don't need to dlopen them.

Tested and working the same as before.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -190,13 +189,6 @@ if(ENABLE_OPTICAL)
   list(APPEND DEP_DEFINES -DHAS_DVD_DRIVE -DHAS_CDDA_RIPPER)
 endif()
 
-if(ENABLE_AIRTUNES)
-  find_package(Shairplay)
-  if(SHAIRPLAY_FOUND)
-    core_require_dyload_dep(Shairplay)
-  endif()
-endif()
```
Comment (Rechi) [inline CMakeLists.txt]: This changes behaviour.

**Review by Rechi — COMMENTED** [review]
You can find the correct changes which removes all unused defines and macros, correct find modules and fixed android packaging at https://github.com/xbmc/xbmc/compare/master...Rechi:cleanup/dlopen.

**Comment by lrusak** [comment]
@Rechi better?

File: CMakeLists.txt [hunk 2]
```diff
@@ -147,24 +147,22 @@ set(optional_deps Alsa
                   MDNS
                   MicroHttpd
                   NFS
+                  Plist
                   PulseAudio
                   Python
+                  Shairplay
```
Comment (Rechi) [inline CMakeLists.txt]: leftover from the `ENABLE_AIRTUNES` removal

File: xbmc/network/AirTunesServer.cpp [hunk 3]
```diff
@@ -696,45 +688,39 @@ bool CAirTunesServer::Initialize(const std::string &password)
...
+  raop_callbacks_t ao = {};
+  ao.cls                  = m_pPipe;
+  ao.audio_init           = AudioOutputFunctions::audio_init;
+  ao.audio_set_volume     = AudioOutputFunctions::audio_set_volume;
+  ao.audio_set_metadata   = AudioOutputFunctions::audio_set_metadata;
+  ao.audio_set_coverart   = AudioOutputFunctions::audio_set_coverart;
+  ao.audio_process        = AudioOutputFunctions::audio_process;
+  ao.audio_destroy        = AudioOutputFunctions::audio_destroy;
+  ao.audio_remote_control_id = AudioOutputFunctions::audio_remote_control_id;
+  ao.audio_set_progress   = AudioOutputFunctions::audio_set_progress;
```
Comment (Rechi) [inline xbmc/network/AirTunesServer.cpp]: remove the vertical alignment in all lines

File: xbmc/network/AirTunesServer.cpp [hunk 4]
```diff
@@ -696,45 +688,39 @@ bool CAirTunesServer::Initialize(const std::string &password)
...
+  m_pRaop = raop_init(1, &ao, RSA_KEY, nullptr); //1 - we handle one client at a time max
+
+  if(m_pRaop)
```
Comment (Rechi) [inline xbmc/network/AirTunesServer.cpp]: space missing between `if` and `(`

File: xbmc/network/AirTunesServer.h [hunk 5]
```diff
@@ -62,8 +63,7 @@ class CAirTunesServer : public ANNOUNCEMENT::IAnnouncer, public IActionListener,
   static void InformPlayerAboutPlayTimes();
 
   int m_port;
-  static DllLibShairplay *m_pLibShairplay;//the lib
-  raop_t *m_pRaop;
+  raop_t *m_pRaop = nullptr;
```
Comment (Rechi) [inline xbmc/network/AirTunesServer.h]: left align `*`

**Review by Rechi — APPROVED** [review]

**Pull request merged**